### ISSUE TITLE
It's ridiculous saving the best model every epoch on Clothing1M

### ISSUE THREAD

**Issue by lxysl** [nazmul-karim170/UNICON-Noisy-Label#12]
```python
# in val() function:
  if acc > best_acc[k-1]:
      best_acc[k-1] = acc
      print('| Saving Best Net%d ...'%k)
      save_point = os.path.join(model_save_loc, '%s_net%d.pth.tar'%(args.id,k))
      torch.save(net.state_dict(), save_point)
```
In each epoch, you load the state_dict from the previous best model and then train it two or three times. If the new model gets a better score, then save it. As all know, LNL methods tend to suffer performance decreases as training goes on, and this is particularly obvious in Clothing1M. So I think it's an unfair comparison unless remove these lines.
